core-java-modules/core-java/src/test/java/com/baeldung/serialization/SerializationUnitTest.java

```diff
@@ -0,0 +1,109 @@
+package com.baeldung.serialization;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.NotSerializableException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.junit.Test;
+
+public class SerializationUnitTest {
+
+	@Test(expected = NotSerializableException.class)
+	public void whenSerializing_ThenThrowsError() throws IOException {
+		Address address = new Address();
+		address.setHouseNumber(10);
+		FileOutputStream fileOutputStream = new FileOutputStream("yofile.txt");
+		try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(fileOutputStream)) {
+			objectOutputStream.writeObject(address);
+		}
+	}
+	
+	@Test
+    public void whenSerializingAndDeserializing_ThenObjectIsTheSame() throws IOException, ClassNotFoundException {
+        Person p = new Person();
+        p.setAge(20);
+        p.setName("Joe");
+
+        FileOutputStream fileOutputStream = new FileOutputStream("yofile.txt");
+		try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(fileOutputStream)) {
+			objectOutputStream.writeObject(p);
+		}
+
+		FileInputStream fileInputStream = new FileInputStream("yofile.txt");
+		try ( ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream)) {
+			Person p2 = (Person) objectInputStream.readObject();
+			assertTrue(p2.getAge() == p.getAge());
```

Comment:
Note that assertEquals(expected, actual) is more conventional in this case. Even though JVM will use the intValue to do the == comparison, it's considered bad practice to use == when comparing objects.

Reply:
Done